### PR TITLE
Fixed requirement on init to have CONDA_NPY set.

### PR DESCRIPTION
**Pull request by pelson**

### CODE REVIEW

File: conda_smithy/cli.py [hunk 1]
```diff
@@ -88,6 +88,13 @@ def __call__(self, args):
                           "conda-recipe you want to build a feedstock for. Got {}".format(
                 args.recipe_directory))
 
+        # conda-build has some really fruity behaviour where it needs CONDA_NPY
+        # and CONDA_PY in order to even read a meta. Because we compute version
+        # matricies anyway the actual number makes absolutely no difference.
+        import conda_build.config
+        conda_build.metadata.config.CONDA_NPY = '99.9'
```
Comment (jakirkham) [inline conda_smithy/cli.py]: Interesting. I guess it really can't. It's not like there are NumPy selectors. Has it done its jinja templating passes by this point?

Comment (pelson) [inline conda_smithy/cli.py]: When it reads the meta, it does two passes. The second pass guarantees that all jinja is expanded at that point. There are most certainly ways around the problem though (as was done by conda-build-all).

Comment (jakirkham) [inline conda_smithy/cli.py]: Right, ok, that's what I was trying to figure out was where this sat in the two pass scheme. Honestly, if we are seeing recipes so crazy that they break this, we probably are not doing a good job reviewing them. ( Famous last words, right. )

Comment (pelson) [inline conda_smithy/cli.py]: *Anything* which depends on numpy was going to have this issue. It was introduced by having more cleverness in the smithy generation code (to skip unnecessary build matrix items).

Comment (jakirkham) [inline conda_smithy/cli.py]: Right, I guess I am trying to think of what new pathological cases could cause this to fail.

**Pull request merged**